src/Asv.Mavlink.Test/Microservices/AsvRadio/AsvRadioClientServerComplexTest.cs

```diff
@@ -0,0 +1,181 @@
+
+        //Act
+        var result = await client.EnableRadio(10000001, AsvRadioModulation.AsvRadioModulationFm, (float)-90.0,
+            (float)90.0,
+            AsvAudioCodec.AsvAudioCodecAac, tcs.Token);
+        await tcs.CancelAsync();
+
+        //Assert
+        Assert.True(result == MavResult.MavResultAccepted);
+        Assert.Equal(AsvRadioCustomMode.AsvRadioCustomModeOnair, server.CustomMode.CurrentValue);
+    }
+
+    [Fact]
+    public async Task Client_EnableRadio_Failure()
+    {
+        //Arrange
+        var tcs = new CancellationTokenSource();
+        var client = Client;
+        var server = Server;
+        server.EnableRadio += async (hz, modulation, dbm, powerDbm, codec, cancel) =>
+        {
+            return await Task.Run(() => MavResult.MavResultAccepted, cancel);
+        };
+        server.Start();
+
+        //Act
+        var result = await client.EnableRadio(10000001, AsvRadioModulation.AsvRadioModulationFm, (float)-100.99,
+            (float)90.0,
+            AsvAudioCodec.AsvAudioCodecAac, tcs.Token);
+
+        //Assert
+        Assert.True(result == MavResult.MavResultFailed);
+        Assert.Equal(AsvRadioCustomMode.AsvRadioCustomModeIdle, server.CustomMode.CurrentValue);
+        await tcs.CancelAsync();
+    }
+
+    [Fact]
+    public async Task Client_DisableRadio_Success()
+    {
+        //Arrange
+        Server.EnableRadio += async (hz, modulation, dbm, powerDbm, codec, cancel) =>
+        {
+            return await Task.Run(() => MavResult.MavResultAccepted, cancel);
+        };
+        Server.DisableRadio += async cancel => { return await Task.Run(() => MavResult.MavResultAccepted, cancel); };
+        Server.Start();
+        var tcs = new CancellationTokenSource();
+
+        //Act
+        ClientTime.Advance(TimeSpan.FromSeconds(1));
+        ServerTime.Advance(TimeSpan.FromSeconds(1));
+        var result = await Client.EnableRadio(10000001, AsvRadioModulation.AsvRadioModulationFm, (float)-99.0,
+            (float)90.0,
+            AsvAudioCodec.AsvAudioCodecAac, tcs.Token);
+
+        //Assert
+        Assert.True(result == MavResult.MavResultAccepted);
+        Assert.Equal(AsvRadioCustomMode.AsvRadioCustomModeOnair, Server.CustomMode.CurrentValue);
+        result = await Client.DisableRadio(tcs.Token);
+        Assert.True(result == MavResult.MavResultAccepted);
+        Assert.Equal(AsvRadioCustomMode.AsvRadioCustomModeIdle, Server.CustomMode.CurrentValue);
+        await tcs.CancelAsync();
+    }
+
+    [Fact]
+    public async Task Client_RequestCodecCapabilities_Success()
+    {
+        //Arrange
+        CreateServer(Identity.Self, ServerCore);
+        var client = Client;
+        var server = Server;
+        var payload = new AsvRadioCodecCapabilitiesResponsePayload();
+        server.Start();
+
+        //Act
+        var result = await client.Base.RequestCodecCapabilities();
+
+        //Assert
+        Assert.Equal(result.Codecs, payload.Codecs);
```

Comment:
it should be compared with _codecs instead of the default payload

Reply:
Request returns array of AsvAudioCodec with 100 of 0U items. "_codecs" is HashSet that delete duplicates. I'd made object same with answer array and compare it. To resolve I'd changed it to simple AsvAudioCodec array that fills at "assert" section of test